Modules/_io/fileio.c

```diff
@@ -447,6 +447,30 @@ _io_FileIO___init___impl(fileio *self, PyObject *nameobj, const char *mode,
         }
     }
     else {
+#if defined(S_ISREG) && defined(ENOTDIR)
+        /* On AIX and Windows, open may succeed for files with a trailing slash.
+           The Open Group specifies filenames ending with a trailing slash should
+           be an error - ENOTDIR */
+	if (S_ISREG(fdfstat.st_mode)) {
+	    int trailing_slash = 0;
+#ifdef MS_WINDOWS
+	    size_t len = wcslen(widename);
```

Comment:
`widename` and `name` can be `NULL` at this point.

Reply:
I cannot imagine how, at this point (as the first call when they must be defined should have returned the error THEN), but yes, since you could conceiveable get to this code without having to call open() - because it was already opened.

will prefix it all that name || widename respectively, is not NULL. If they are, they must have already passed successfully.

Reply:
@aixtools, Python's `open` function accepts an integer FD instead of a filename. See my comment above. There's no need to check `name` and `widename` separately. Just check the common `fd` value.

Reply:
tomorrow then. But if there is a fd already - should I care? assume the file was opened by an extern module. If that has accepted  trailing slash, I, as python, no longer care. I, python, only care when I am doing an open() using a (wide)name.

Reply:
You've misunderstood me. The `fd` variable is not `self->fd`. Please re-read the function from the top. It's assigned as `fd = _PyLong_AsInt(nameobj)`. If Python `open` was called with an integer fd, then this variable will be 0 or greater. If it's less than 0, and we've made it this far, then we know that `open` was called with a filename, not an `fd`. This is all you have to test. You do not have to check whether `name` or `widename` is `NULL`.